packages/flutter_tools/lib/src/devtools_launcher.dart

```diff
@@ -84,7 +93,11 @@ class DevtoolsServerLauncher extends DevtoolsLauncher {
         }
       );
 
-      devToolsUrl = await completer.future;
+      // We do not need to wait for a [Completer] holding the DTD URI because
+      // the DTD URI will be output to stdout before the DevTools URI. Awaiting
```

Comment:
Is this an implementation detail that could be inadvertently broken in a future refactor?

Reply:
Could also be a reason to use the nifty new FutureRecord2 extension method .wait() :)

https://api.dart.dev/stable/3.3.0/dart-async/FutureRecord2/wait.html

Reply:
The DTD uri won't always be available here, so we can't await a completer with the guarantee that it will complete. I can add tests in https://dart-review.googlesource.com/c/sdk/+/354460 that verify the order, with a comment that if the order or strings should change, that we will need to update the Flutter tools launcher. We could already get in a broken state with the existing "Serving DevTools" string if that were ever to change.